tools/wheels/cibw_before_build_macos.sh

```diff
@@ -8,52 +8,18 @@ echo $PLATFORM
 cat $PROJECT_DIR/tools/wheels/LICENSE_osx.txt >> $PROJECT_DIR/LICENSE.txt
 
 #########################################################################################
-# Install GFortran + OpenBLAS
```

Comment:
After switching to the pre-installed `gfortran`, I noticed warnings in the [x86 OpenBLAS build logs](https://github.com/thalassemia/scipy/actions/runs/8710829335/job/23893586497) like:

    ld: warning: dylib (/usr/local/Cellar/gcc/13.2.0/lib/gcc/current/libgfortran.dylib) was built for newer macOS version (13.0) than being linked (10.9)

This is because we set `MACOSX_DEPLOYMENT_TARGET=10.9`. I'm not sure why this even works when [the lowest supported deployment target](https://developer.apple.com/support/xcode/) for Xcode 15.2 is macOS 10.13, but regardless I was able to [install and run the built wheel](https://github.com/thalassemia/test-scipy-wheels/actions/runs/8727212293/job/23944241610) on a macOS 11 runner just fine (not that old but older than the 13.0 from the warnings).

Reply:
After thinking this over some more, I realized I'm probably playing with fire here. The `gfortran` packages we pulled in this script do have the correct deployment targets (10.9 for x86 and 12.0 for ARM). I'll revert this change.

Reply:
I think it's fine not to install gfortran here, we'll use the gfortran on the stock runner image.

Reply:
After reverting this file, I found out that the ARM `gfortran` we pull actually targets macOS 12.2, so we still get warnings like the above because our ARM wheels target macOS 12.0. To silence the build warnings, I set `MACOSX_DEPLOYMENT_TARGET=12.2`, which `cibuildwheel` converts to `12_0` in the final wheel name. With this, the only users who might run into weird edge case behavior are those with 12.0 <= macOS < 12.2.

Reply:
Ouch, lots of compiler-related build failures. Trying to work with the custom `gfortran` compilers is a real pain. I'll go back to using the installed compilers and let others chime in about whether this warning is worth acting on.

Reply:
No strong preference on this - there is no ideal solution. The custom-built ones are more stable while the GHA-installed one can be replaced and break at any time; on the other hand the GHA-installed one is easier to deal with so we can also deal with any breakage if and when it occurs.

Reply:
If we delete this, please update the toolchain.rst file accordingly